GlobalstatsIO.cs

```diff
@@ -5,318 +5,315 @@
+        {
+            public string name = null;
+            public string _id = null;
+
+            [SerializeField]
+            public List<StatisticValues> values = null;
+        }
+
+        private AccessToken api_access_token = null;
+
+        private List<StatisticValues> statistic_values = new List<StatisticValues>();
+
+        [HideInInspector]
+        public string statistic_id = "";
+
+        [HideInInspector]
+        public string user_name = "";
+
+        [HideInInspector]
+        public LinkData link_data = null;
+
+        private IEnumerator GetAccessToken()
+        {
+            if (api_id == "" || api_secret == "")
+            {
+                throw new Exception("Credentials missing");
+            }
+
+            string url = "https://api.globalstats.io/oauth/access_token";
+
+            WWWForm form = new WWWForm();
+            form.AddField("grant_type", "client_credentials");
+            form.AddField("scope", "endpoint_client");
+            form.AddField("client_id", api_id);
+            form.AddField("client_secret", api_secret);
+
+            UnityWebRequest www = UnityWebRequest.Post(url, form);
+            www.downloadHandler = new DownloadHandlerBuffer();
+            yield return www.SendWebRequest();
+
+            string responseBody = www.downloadHandler.text;
+
+            if (www.isNetworkError || www.isHttpError)
+            {
+                Debug.LogWarning("Error retrieving access token: " + www.error);
+                Debug.Log("GlobalstatsIO API Response: " + responseBody);
+                yield break;
+            }
+            else
+            {
+                api_access_token = JsonUtility.FromJson<AccessToken>(responseBody);
+            }
+        }
+
+        public IEnumerator Share(string id = "", string name = "", Dictionary<string, string> values = null, Action<bool> callback = null)
+        {
+            bool update = false;
+
+            if (api_access_token == null || !api_access_token.IsValid())
+            {
+                yield return GetAccessToken();
+            }
+
+            // If no id is supplied but we have one stored, reuse it.
+            if (id == "" && statistic_id != "")
+                id = statistic_id;
+
+            string url = "https://api.globalstats.io/v1/statistics";
+            if (id != "")
+            {
+                url = "https://api.globalstats.io/v1/statistics/" + id;
+                update = true;
+            }
+            else
+            {
+                if (name == "")
+                    name = "anonymous";
+            }
+
+            string json_payload;
+
+            if (update == false)
+            {
+                json_payload = "{\"name\":\"" + name + "\", \"values\":{";
+            }
+            else
+            {
+                json_payload = "{\"values\":{";
+            }
+
+            bool semicolon = false;
+            foreach (KeyValuePair<string, string> value in values)
```

Comment:
You've made `value` optional in the method signature: `Dictionary<string, string> values = null`
This will throw an exception if the method is called without any values.
TBH, I don't think it makes much sense to call the method without `values` so I suggest to remove the default argument.